frontend/public/components/machine-config.tsx

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable no-undef, no-unused-vars */
+import * as _ from 'lodash-es';
+import * as React from 'react';
+
+import { MachineConfigKind, referenceForModel } from '../module/k8s';
+import { MachineConfigModel } from '../models';
+import { fromNow } from './utils/datetime';
+import {
+  ColHead,
+  DetailsPage,
+  List,
+  ListHeader,
+  ListPage,
+} from './factory';
+import {
+  Kebab,
+  navFactory,
+  ResourceKebab,
+  ResourceLink,
+  ResourceSummary,
+  SectionHeading,
+} from './utils';
+
+
+export const machineConfigReference = referenceForModel(MachineConfigModel);
+const machineConfigMenuActions = [...Kebab.factory.common];
+
+const MachineConfigSummary: React.SFC<MachineConfigSummaryProps> = ({obj}) => (
+  <ResourceSummary resource={obj} showPodSelector={false} showNodeSelector={false}>
+    <dt>OS Image URL</dt>
+    <dd>{obj.spec.osImageURL || '-'}</dd>
+  </ResourceSummary>
+);
+
+const MachineConfigDetails: React.SFC<MachineConfigDetailsProps> = ({obj}) => (
+  <div className="co-m-pane__body">
+    <SectionHeading text="Machine Config Overview" />
+    <div className="row">
+      <div className="col-xs-12">
+        <MachineConfigSummary obj={obj} />
+      </div>
+    </div>
+  </div>
+);
+
+const pages = [
+  navFactory.details(MachineConfigDetails),
+  navFactory.editYaml(),
+];
+
+export const MachineConfigDetailsPage: React.SFC<any> = props => {
+  return <DetailsPage {...props} kind={machineConfigReference} menuActions={machineConfigMenuActions} pages={pages} />;
+};
+
+const MachineConfigHeader: React.SFC<any> = props => <ListHeader>
+  <ColHead {...props} className="col-xs-6  col-sm-4  col-md-3  col-lg-2" sortField="metadata.name">Name</ColHead>
+  <ColHead {...props} className="hidden-xs col-sm-6  col-md-4  col-lg-3" sortField="metadata.annotations.machineconfiguration.openshift.io/generated-by-controller-version">Generated By Controller</ColHead>
```

Comment:
I don't think this `sortField` will work. I'd have to test, but I think you can use `[]`

```suggestion
  <ColHead {...props} className="hidden-xs col-sm-6  col-md-4  col-lg-3" sortField="metadata.annotations.machineconfiguration['openshift.io/generated-by-controller-version']">Generated By Controller</ColHead>
```

Reply:
fixed